ltsp/server/ipxe/55-ipxe.sh

```diff
@@ -28,46 +37,66 @@ ipxe_cmdline() {
 }
 
 ipxe_main() {
-    local key items gotos r_items r_gotos img_name title client_sections binary
+    local key items gotos methods boot_name boot_names item_label title client_sections binary sedp subdir boot_method_vars boot_method_var boot_method boot_method_name boot_method_target is_default
 
-    # Prepare the menu text for all images and chroot
+    # Prepare the menu text for all images and chroots
     key=0
-    items=""
-    gotos=":images"
-    img_name=$(re list_img_names -i)
-    set -- $img_name
-    for img_name in "$@"; do
-        key=$((key+1))
-        title=$(echo_values "$(ipxe_name "$img_name.img")")
-        title=${title:-$img_name.img}
-        items="${items:+"$items\n"}$(printf "item --key %d %-20s %s" "$((key%10))" "$img_name" "$title")"
-        gotos=":$img_name\n$gotos"
-    done
-    r_items=""
-    r_gotos=":roots"
-    img_name=$(re list_img_names -c)
-    set -- $img_name
-    for img_name in "$@"; do
+    items=""   # The menu items to be inserted.
+    gotos=""   # Jump targets setting correct ${img} var for menu items.
+    methods="" # Jump targets with the respective boot_method.
+
+    # INFO: The images/chroots automatism has been removed. It has been replaced
+    # by the BOOT_METHOD functionality, which can achieve the same thing but
+    # without hardwiring LTSP to any specific BOOT_METHOD.
+    # As a side effect some code repititions have also been remove.
+
+    # Extracts all vars with pattern BOOT_METHOD_[0-9]* and prepares boot
+    # options for their respective dirs.
+    boot_method_vars=$(echo_vars "BOOT_METHOD_[0-9]*")
+    set -- $boot_method_vars
+
+    for boot_method_var in "$@"; do
+      eval "subdir=\$${boot_method_var}_DIR"
+      eval "boot_method=\$${boot_method_var}"
+      eval "boot_method_name=\$${boot_method_var}_NAME"
+      boot_method_target=$(ipxe_lowercase ${boot_method_var})
+      methods="${methods}${boot_method_target}:\nset cmdline_boot_method $boot_method \&\& goto ltsp\n"
```

Comment:
Small bug with the jump target name:

Needs to reed like this
methods="${methods}:${boot_method_target}\nset cmdline_boot_method $boot_method \&\& goto ltsp\n"